errortypes/code.go

```diff
@@ -0,0 +1,36 @@
+package errortypes
+
+// Defines numeric codes for non-specific errors or warnings.
+const (
+	UnknownCode = 999
+)
+
+// Defines numeric codes for well-known errors.
+const (
+	TimeoutCode = iota + 1
+	BadInputCode
+	BlacklistedAppCode
+	BadServerResponseCode
+	FailedToRequestBidsCode
+	BidderTemporarilyDisabledCode
+	BlacklistedAcctCode
+	AcctRequiredCode
+)
+
+// Defines numeric codes for well-known warnings.
+const (
+	InvalidPrivacyConsentWarningCode = iota + 10001
+)
+
+// Coder provides an error or warning code.
+type Coder interface {
+	Code() int
+}
+
+// ReadErrorCode returns the error or warning code, or UnknownCode if unavailable.
+func ReadErrorCode(err error) int {
+	if e, ok := err.(Coder); ok {
+		return e.Code()
+	}
+	return UnknownCode
```

Comment:
I would need to add a Severity check here as well, or just combine the ErrorCode and SeverityLevel methods into a single CodedError interface.. which might honestly make sense.

Reply:
I could see having a method that just returns a touple to deliver both values.